src/main/java/org/prebid/server/bidder/decenterads/DecenteradsBidder.java

```diff
@@ -0,0 +1,129 @@
+package org.prebid.server.bidder.decenterads;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Decenterads {@link Bidder} implementation.
+ */
+public class DecenteradsBidder implements Bidder<BidRequest> {
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public DecenteradsBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<HttpRequest<BidRequest>> validRequests = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final JsonNode resolvedImpExt = resolveImpExt(imp.getExt());
+                final Imp updatedImp = updateImp(imp, resolvedImpExt);
+
+                validRequests.add(createRequest(request, updatedImp));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        return Result.of(validRequests, errors);
+    }
+
+    private static JsonNode resolveImpExt(ObjectNode impExt) {
+        final JsonNode newImExt = impExt.get("bidder");
+        if (newImExt == null || newImExt.isEmpty()) {
+            throw new PreBidException("bidder parameters required");
```

Comment:
Pls check do we really need this because of `org.prebid.server.validation.RequestValidator#validateImpBidderExtName` do something similar.

Reply:
org.prebid.server.validation.RequestValidator#validateImpExtPrebidBidder calls org.prebid.server.validation.RequestValidator#validateImpBidderExtName, but before call it has `final JsonNode extPrebidBidder = extPrebid.get(BIDDER_EXT);

        if (extPrebidBidder == null) {
            return;
        }`
So seems it is usefull check, am I wrong?

Reply:
Actually, i'm talking about `org.prebid.server.validation.RequestValidator#validateImpExtPrebid` which throws `request.imp[%d].ext.prebid must be non-empty object` exception.

Btw, pls change the obtaining extension from `imp[].ext.prebid.bidder` according to the last changes.

Reply:
Found that we already have null check for extPrebidBidderNode, so removed `newImExt == null `